release/serve_tests/workloads/single_deployment_1k_noop_replica.py

```diff
@@ -0,0 +1,196 @@
+
+# Cluster setup configs
+NUM_REDIS_SHARDS = 1
+REDIS_MAX_MEMORY = 10**8
+OBJECT_STORE_MEMORY = 10**8
+NUM_CPU_PER_NODE = 8
+
+# Deployment configs
+DEFAULT_MAX_BATCH_SIZE = 16
+
+# Experiment configs - wrk specific
+DEFAULT_SMOKE_TEST_TRAIL_LENGTH = "1m"
+DEFAULT_FULL_TEST_TRAIL_LENGTH = "10m"
+
+
+def setup_local_single_node_cluster(num_nodes):
+    """Setup ray cluster locally via ray.init() and Cluster()
+
+    Each actor is simulated in local process on single node,
+    thus smaller scale by default.
+    """
+    cluster = Cluster()
+    for i in range(num_nodes):
+        cluster.add_node(
+            redis_port=6379 if i == 0 else None,
+            num_redis_shards=NUM_REDIS_SHARDS if i == 0 else None,
+            num_cpus=NUM_CPU_PER_NODE,
+            num_gpus=0,
+            resources={str(i): 2},
+            object_store_memory=OBJECT_STORE_MEMORY,
+            redis_max_memory=REDIS_MAX_MEMORY,
+            dashboard_host="0.0.0.0",
+        )
+    ray.init(address=cluster.address, dashboard_host="0.0.0.0")
+    serve.start()
+
+
+def setup_anyscale_cluster():
+    """Setup ray cluster at anyscale via ray.client()
+
+    Note this is by default large scale and should be kicked off
+    less frequently.
+    """
+    # TODO: Ray client didn't work with releaser script yet because
+    # we cannot connect to anyscale cluster from its headnode
+    # ray.client().env({}).connect()
+    ray.init(address="auto")
+    serve.start()
+
+
+def deploy_replicas(num_replicas, max_batch_size):
+    @serve.deployment(name="echo", num_replicas=num_replicas)
+    class Echo:
+        @serve.batch(max_batch_size=max_batch_size)
+        async def handle_batch(self, requests):
+            return ["hi" for _ in range(len(requests))]
+
+        async def __call__(self, request):
+            return await self.handle_batch(request)
+
+    Echo.deploy()
+
+
+def warm_up_cluster(num_warmup_iterations: int) -> None:
+    for _ in range(num_warmup_iterations):
+        resp = requests.get("http://127.0.0.1:8000/echo").text
+        print(resp)
+        time.sleep(0.5)
+
+
+def run_one_trial(trail_length: str, num_connectionss) -> None:
+    proc = subprocess.Popen(
+        [
+            "wrk",
+            "-c",
+            str(num_connectionss),
+            "-t",
+            str(NUM_CPU_PER_NODE),
+            "-d",
+            trail_length,
+            "http://127.0.0.1:8000/echo",
+        ],
+        stdout=PIPE,
+        stderr=PIPE,
+    )
+    proc.wait()
+    out, err = proc.communicate()
+    print(out.decode())
+    print(err.decode())
+
+    return out.decode()
+
+
+def shutdown_cluster():
+    pass
+
+
+@click.command()
+@click.option("--num-replicas", type=int)
+@click.option("--num-trials", type=int)
+@click.option("--trial-length", type=str)
+@click.option("--max-batch-size", type=int, default=DEFAULT_MAX_BATCH_SIZE)
+@click.option("--run-locally", type=bool, default=True)
+@click.option("--smoke-test", type=bool, default=True)
```

Comment:
default to false, and we actually use env var `IS_SMOKE_TEST=0` in releaser, so try use that instead?